app/test/unit/main-process/spell-checker-menu-test.ts

```diff
@@ -0,0 +1,58 @@
+import { getSpellCheckLanguageMenuItemOptions } from '../../../src/main-process/menu/build-spell-check-menu'
+import { getAvailableSpellcheckerLanguages } from '../../helpers/menus/available-spellchecker-languages-helper'
+
+describe('spell-checker-menu', () => {
+  it('spell checker language is english && user language is english', () => {
+    const spellcheckerLanguage = ['en-US']
+    const userLanguageCode = 'en-US'
+
+    const spellcheckerMenuItem = getSpellCheckLanguageMenuItemOptions(
+      userLanguageCode,
+      spellcheckerLanguage,
+      getAvailableSpellcheckerLanguages()
+    )
+
+    expect(spellcheckerMenuItem).toBeNull()
+  })
+
+  it('spell checker language is english && user language is not english && user language is not supported', () => {
+    const spellcheckerLanguage = ['en-US']
+    const userLanguageCode = 'zh-CN'
+
+    const spellcheckerMenuItem = getSpellCheckLanguageMenuItemOptions(
+      userLanguageCode,
+      spellcheckerLanguage,
+      getAvailableSpellcheckerLanguages()
+    )
+
+    expect(spellcheckerMenuItem).toBeNull()
+  })
+
+  it('spell checker language is english && user language is not english && user language is supported', () => {
+    const spellcheckerLanguage = ['en-US']
+    const userLanguageCode = 'bg'
+
+    const spellcheckerMenuItem = getSpellCheckLanguageMenuItemOptions(
+      userLanguageCode,
+      spellcheckerLanguage,
+      getAvailableSpellcheckerLanguages()
+    )
+
+    expect(spellcheckerMenuItem?.label).toBe(
+      'Set spellcheck to system language'
+    )

```

Comment:
Not extremely concerned about this, but I think it's better if we don't use specific user-facing strings in tests. Maybe we could do something like…
```suggestion
    expect(spellcheckerMenuItem?.label).not.toBeFalsy()
```

That will check the label is not undefined, nor null, nor an empty string.

Reply:
I am not sure I agree with this one. I get the idea - but, the label is the result we are checking for. There could be 2 different labels or null. If we want the test to be agnostic, we could put the labels in an exportable constant used by the implementation and the test.

Reply:
I refactor the tests and impl to rely on constants as @tidy-dev suggested!

Reply:
Sounds good, thank you both! ❤️